src/main/java/org/betonquest/betonquest/compatibility/traincarts/TrainCartsUtils.java

```diff
@@ -0,0 +1,89 @@
+import org.bukkit.entity.Entity;
+
+import javax.annotation.Nullable;
+
+/**
+ * Utility class for TrainCarts.
+ */
+public final class TrainCartsUtils {
+
+    /**
+     * This class cannot be instantiated.
+     */
+    private TrainCartsUtils() {
+    }
+
+    /**
+     * Checks if a profile is currently riding a train from TrainCarts with the specified name.
+     *
+     * @param profile   the {@link OnlineProfile} to check.
+     * @param trainName the name of the train.
+     * @return {@code true} if the player is riding the train with the specified name, otherwise {@code false}.
+     */
+    public static boolean isRidingTrainCart(final OnlineProfile profile, final String trainName) {
+        final MinecartMember<?> minecartMember = getMinecartMember(profile);
+        if (minecartMember == null) {
+            return false;
+        }
+
+        final String groupName = minecartMember.getGroup().getProperties().getTrainName();
+        return trainName.equals(groupName);
+    }
+
+    /**
+     * Checks if a profile is currently riding a train from TrainCarts.
+     *
+     * @param profile the {@link OnlineProfile} to check.
+     * @return {@code true} if the player is riding a train, otherwise {@code false}.
+     */
+    public static boolean isRidingTrainCart(final OnlineProfile profile) {
+        return getMinecartMember(profile) != null;
+    }
+
+    /**
+     * Gets the {@link MinecartMember} the player is currently riding.
+     *
+     * @param profile the {@link OnlineProfile} to get the {@link MinecartMember} from.
+     * @return the {@link MinecartMember} the player is currently riding or else {@code null}.
+     */
+    @Nullable
+    public static MinecartMember<?> getMinecartMember(final OnlineProfile profile) {
+        final Entity vehicle = profile.getPlayer().getVehicle();
+        if (vehicle == null) {
+            return null;
+        }
+        return MinecartMemberStore.getFromEntity(vehicle);
+    }
+
+    /**
+     * Checks if the train Name matches with the name from the instruction.
+     *
+     * @param log           the {@link BetonQuestLogger} to log upcoming errors.
+     * @param pack          the {@link QuestPackage} of the instruction.
+     * @param packId        the {@link ID} of the instruction.
+     * @param onlineProfile the {@link OnlineProfile} to get the value from the variable.
+     * @param name          the {@link VariableString} name of the train from the instruction.
+     * @param trainName     the name of the train to check.
+     * @return {@code true} if the train name matches with the name from the instruction, otherwise {@code false}.
+     */
+    public static boolean isValidTrain(final BetonQuestLogger log, final QuestPackage pack, final ID packId,
+                                       final OnlineProfile onlineProfile, final VariableString name, final String trainName) {
+        try {
+            final String result = name.getValue(onlineProfile);
```

Comment:
The value get should not be part of the Util class in my opinion.
You also don't need a logger when you just "normal" throw in the methods.
Edit: This can be useful for the EventHandler, yes. I don't know what the best way for this is under that aspect.

Reply:
I am also not sure about that `""` is valid for everything.

Reply:
Its a own method to remove duplicated code. With this aspect I would let this method like it is.